compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -1460,28 +1461,21 @@ private STNode parseFuncDefinition(STNode metadata, boolean isObjectMethod, STNo
      * </code>
      *
      * @param metadata Metadata
-     * @param visibilityQualifier Visibility qualifier
+     * @param qualifiers qualifier list
      * @return Parsed node
      */
-    private STNode parseFuncDefOrFuncTypeDesc(STNode metadata, boolean isObjectMethod, STNode... qualifiers) {
-        parseTransactionalQUalifier(qualifiers);
+    private STNode parseFuncDefOrFuncTypeDesc(STNode metadata, boolean isObjectMethod, List<STNode> qualifiers) {
+        if (peek().kind == SyntaxKind.TRANSACTIONAL_KEYWORD) {
```

Comment:
If the token is neither `transactional` or `function`, we'll have to add recovery here

Reply:
Added.